### PR TITLE
fix: #916  added profile/id route

### PR DESCRIPTION
**Pull request by codexdelta**
Related issue: #916 

Summary: Change user's profile to use /profile/[id]. 
Allowing the user to share the profile url with anyone, with its unique profileId.

### CODE REVIEW

**Comment by ryanwarsaw** [comment]
Related: #916 

Hey there, two things I noticed. In the future, you can reference issues by typing "#" followed by the issue number within the body of your Pull Request, GitHub will automatically link it if the issue exists. It helps reviewers keep track of what issues correlate to PRs, and generally is good house keeping.

Also, you may have noticed that your Git commits don't show your GitHub avatar, rather they show a grayed out GitHub logo. This is because the email you have in your git settings, and the one you have associated with your GitHub account, are not the same. If pair them together, your contributions will link to your profile (so you can keep track of all your work 🎉 ).

**Comment by xmatthewx** [comment]
Excellent. This works as expected from the perspective of user experience. 

But, I will let @mmmavis or others review the code before we merge.

**Comment by mmmavis** [comment]
Hey @codexdelta is this PR ready for review? If so, let's remove "WIP" from the PR title 👍 .

**Comment by codexdelta** [comment]
Hi @mmmavis, it's not ready for review. It's still in WIP, I will tag you for review once it's done. 
I have always found having early PR with WIP helps in early feedback.

**Comment by mmmavis** [comment]
@codexdelta It seems we are hoping to have this fix in soon to unblock some of our colleagues' work (correct, @xmatthewx ?) I'm gonna have a quick look at your WIP PR now. Let me know if anything needs more clarification. 😃

File: js/app-user.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ const Login = {
     // verify user's logged in status with Pulse API
     Service.userstatus()
       .then(response => {
-        setUserData(false, response.username, response.customname, response.email, response.moderator);
+        setUserData(false, response.username, response.profileid, response.customname, response.email, response.loggedin, response.moderator);
```
Comment (mmmavis) [inline js/app-user.js]: There are way too many arguments here. We should probably just pass in `response` object here and fix other functions correspondingly (maybe change the `update` function parameters to `update(error, user)`)

**Comment by codexdelta** [comment]
The current PR will break if the user goes to another profile from his own profile by changing url and comes back to his own profile by clicking the navlink with his name. I am looking for the fix or the right way to do it. 

It would be great if you look in at my-profile.jsx and suggest me what am I missing.

**Comment by xmatthewx** [comment]
if you have a moment @mmmavis @alanmoo, let's review this. there will be a bunch of new people and profiles around fellows and global sprint that would benefit from this.

File: main.jsx [hunk 2]
```diff
@@ -17,7 +17,6 @@ import Submitted from './pages/add/submitted.jsx';
 import Search from './pages/search/search.jsx';
 import Moderation from './pages/moderation.jsx';
 import PublicProfile from './pages/public-profile.jsx';
-import MyProfile from './pages/my-profile.jsx';
```
Comment (mmmavis) [inline main.jsx]: we should delete this file this it's not being used anywhere anymore

**Comment by mmmavis** [comment]
`<Link to="/profile/me">` in `entry.jsx` needs to be updated too since there's no longer a `/profile/me` route

File: main.jsx (hunk 2 above)
Comment (codexdelta) [reply main.jsx]: done, please check in with latest squashed commits.

File: js/app-user.js (hunk 1 above)
Comment (codexdelta) [reply js/app-user.js]: done, please check in with latest squashed commits.

File: js/app-user.js [hunk 3]
```diff
@@ -77,11 +77,20 @@ class User {
   constructor() {
     this.listeners = [];
     this.resetUser();
+    this.defaultUserData = {
```
Comment (mmmavis) [inline js/app-user.js]: This should probably happen in `resetUser()`

Comment (mmmavis) [reply js/app-user.js]: this comment here needs some updates https://github.com/mozilla/network-pulse/pull/926/files#diff-c63ea7b512f87e9b9c8728136d59562dR20

Comment (codexdelta) [reply js/app-user.js]: if any of the fields of the user is not present then it should be set to false. Am I assuming it right? 

and for the comment, you mentioned we can show a logout notification when the user logs out successfully.

File: js/app-user.js [hunk 4]
```diff
@@ -136,25 +145,25 @@ class User {
...
-    this.moderator = !!moderator;
-    this.name = customName || username; // use user's custom name if it exists. otherwise fall back to user's user name
-    this.email = email;
-
+    this.loggedin = !!userData.loggedin;
+    this.moderator = !!userData.moderator;
+    this.name = userData.customName || userData.username; // use user's custom name if it exists. otherwise fall back to user's user name
+    this.email = userData.email;
+    this.profileid = userData.profileid;
     // notify listeners that this user logged in state has been verified
```
[the remark this answers is not in the document]
Comment (codexdelta) [reply js/app-user.js]: done, I will be very careful with all the future PRs.

[the remark this answers is not in the document]
Comment (mmmavis) [reply js/app-user.js]: 👍

[the remark this answers is not in the document]
Comment (mmmavis) [reply js/app-user.js]: @codexdelta I'm not seeing the extra line here. Mind making sure you git committed and pushed your code change?

**Comment by mmmavis** [comment]
Hey @codexdelta could you address all issues that I mentioned and tag me for review when that's done? It's a bit hard for me to keep up with your WIP code, unaddressed issues, and threads of comments at the same time. 😬

**Comment by codexdelta** [comment]
It is ready for review with all the requested changes, @mmmavis  

Thanks

**Comment by alanmoo** [comment]
I tested this and it seems to work great! Code looks good too; Mavis if you want to give it another look today go ahead; Otherwise I'll merge it in on Monday.

**Comment by codexdelta** [comment]
@mmmavis  Thank you for your guidance. 🙂

**Pull request merged**